transform.js

```diff
@@ -8,125 +8,172 @@ var logger = require('./logger');
-    return offset;
-  };
+var guessPumpOffset = (function () {
+    var lastGuess;
+
+    // From my observations, sMedicalDeviceTime is advanced by the server even when the app is
+    // not reporting data or the pump is not connected, so its difference from server time is
+    // always close to a whole number of hours, and can be used to guess the pump's timezone:
+    // https://gist.github.com/mddub/f673570e6427c93784bf
+    return function (data) {
+        var pumpTimeAsIfUTC = Date.parse(data['sMedicalDeviceTime'] + ' +0');
+        var serverTimeUTC = data['currentServerTime'];
+        var hours = Math.round((pumpTimeAsIfUTC - serverTimeUTC) / (60 * 60 * 1000));
+        var offset = (hours >= 0 ? '+' : '-') + (Math.abs(hours) < 10 ? '0' : '') + Math.abs(hours) + '00';
+        if (offset !== lastGuess) {
+            logger.log('Guessed pump timezone ' + offset + ' (pump time: "' + data['sMedicalDeviceTime'] + '"; server time: ' + new Date(data['currentServerTime']) + ')');
+        }
+        lastGuess = offset;
+        return offset;
+    };
 })();
 
 function deviceStatusEntry(data, offset) {
-  return {
-    'created_at': timestampAsString(data['lastMedicalDeviceDataUpdateServerTime']),
-    'device': deviceName(data),
-    'uploader': {
-      'battery': data['conduitBatteryLevel'],
-    },
-    'pump': {
-      'battery': {
-        'percent': data['medicalDeviceBatteryLevelPercent'],
-      },
-      'reservoir': data['reservoirAmount'],
-      'iob': {
-        'timestamp': timestampAsString(data['lastMedicalDeviceDataUpdateServerTime']),
-        'bolusiob': _.get(data, 'activeInsulin.amount') >= 0 ? _.get(data, 'activeInsulin.amount') : undefined,
-      },
-      'clock': timestampAsString(parsePumpTime(data['sMedicalDeviceTime'], offset)),
-      // TODO: add last alarm from data['lastAlarm']['code'] and data['lastAlarm']['datetime']
-      // https://gist.github.com/mddub/a95dc120d9d1414a433d#file-minimed-connect-codes-js-L79
-    },
-    'connect': {
-      // For the values these can take, see:
-      // https://gist.github.com/mddub/5e4a585508c93249eb51
-      'sensorState': data['sensorState'],
-      'calibStatus': data['calibStatus'],
-      'sensorDurationHours': data['sensorDurationHours'],
-      'timeToNextCalibHours': data['timeToNextCalibHours'],
-      'conduitInRange': data['conduitInRange'],
-      'conduitMedicalDeviceInRange': data['conduitMedicalDeviceInRange'],
-      'conduitSensorInRange': data['conduitSensorInRange'],
+    if (data['medicalDeviceFamily'] === 'GUARDIAN') {
+        return {
+            'created_at': timestampAsString(data['lastMedicalDeviceDataUpdateServerTime']),
+            'device': deviceName(data),
+            'uploader': {
+                'battery': data['medicalDeviceBatteryLevelPercent'],
+            },
+            'guardian': {
+                // For the values these can take, see:
+                // https://gist.github.com/mddub/5e4a585508c93249eb51
```

Comment:
This comment doesn't apply here, since that Gist is for Connect data only.

Reply:
I adapted the trends to Nightscout. When Carelink shows DOWN_TRIPLE and UP_TRIPLE - what then?